File: jar-common/src/main/java/com/beancounter/common/contracts/TrnResponse.java
package com.beancounter.common.contracts;

import com.beancounter.common.model.Trn;
import java.util.ArrayList;
import java.util.Collection;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class TrnResponse implements Payload<Collection<Trn>> {
  @Builder.Default
  private Collection<Trn> data = new ArrayList<>();
}

File: jar-common/src/main/kotlin/com/beancounter/common/contracts/TrnResponse.kt
package com.beancounter.common.contracts

import com.beancounter.common.model.Trn

/**
 * Response envelope for trn queries. [data] is a [TrnPayload] that
 * de-duplicates the Asset / Portfolio / Currency / Broker objects shared
 * by the contained trns.
 */
data class TrnResponse(
    override val data: TrnPayload = TrnPayload()
) : Payload<TrnPayload> {
    constructor(trns: Collection<Trn>) : this(TrnPayload.from(trns))
}